docs/decisions/0049-agents-assistantsV2.md

```diff
@@ -0,0 +1,157 @@
+
+<p align="center">
+<kbd><img src="diagrams/assistant-threadcreationsettings.png" style="width: 132pt;"></kbd>
+</p>
+
+
+#### Service Configuration
+
+The `OpenAIServiceConfiguration` defines how to connect to a specific remote service, whether it be OpenAI, Azure, or proxy.  This eliminates the need to define multiple overloads for each call site that results in a connection to the remote API service (i.e. create a _client)_.
+
+> Note: This was previously named `OpenAIAssistantConfiguration`, but is not necessarily assistant specific.
+
+<p align="center">
+<kbd><img src="diagrams/assistant-serviceconfig.png"  style="width: 360pt;"></kbd>
+</p>
+
+
+### Agent Implementation
+
+The `OpenAIAssistant` agent is roughly equivalent to its V1 form save for:
+
+- Supports settings for _assistant_, _thread_, and _run_
+- Agent definition shifts to `Definition` property
+
+<p align="center">
+<kbd><img src="diagrams/assistant-agent.png"  style="width: 720pt;"></kbd>
+</p>
+
+The following table describes the purpose of diagramed methods on the `OpenAIAssistantAgent`.
+
+|Method Name|Description|
+---|---
+**Create**|Create a new assistant agent
+**ListDefinitions**|List existing assistant definitions
+**Retrieve**|Retrieve an existing assistant
+**CreateThread**|Create an assistant thread
+**DeleteThread**|Delete an assistant thread
+**AddChatMessage**|Add a message to an assistant thread
+**GetThreadMessages**|Retrieve all messages from an assistant thread
+**Delete**|Delete the assistant agent's definition (puts agent into a terminal state)
+**Invoke**|Invoke the assistant agent (no chat)
+**GetChannelKeys**|Inherited from `Agent`
+**CreateChannel**|Inherited from `Agent`
+
+
+### Vector Store Support
+
+_Vector Store_ support is required in order to enable usage of the `file-search` tool.  
+
+In alignment with V2 treaming of the `FileClient`, the caller may also directly target `VectorStoreClient` from the _OpenAI SDK_.
+
+
+### Class Inventory
+This section provides an overview / inventory of all the public surface area described in this ADR.
+
+|Class Name|Description|
+---|---
+**OpenAIAssistantAgent**|An `Agent` based on the Open AI Assistant API
+**OpenAIAssistantChannel**|An 'AgentChannel' for `OpenAIAssistantAgent` (associated with a _thread-id_.)
+**OpenAIAssistantDefinition**|All of the metadata / definition for an Open AI Assistant.  Unable to use the _Open AI API_ model due to implementation constraints (constructor not public).
+**OpenAIAssistantExecutionSettings**|Setting that affect the _run_, but defined globally for the agent/assistant.
+**OpenAIAssistantInvocationSettings**|Settings bound to a discrete run, used for direct (no chat) invocation.
+**OpenAIServiceConfiguration**|Describes the service connection and used to create the `OpenAIClient`
+
+### Run Processing
```

Comment:
I love this section and the high level overview about what it takes to run the OpenAI Assistant agent. One thing I noticed as I look into the Python OpenAI assistants V2 library is that it is mainly built to accept keyword-arguments, and not specific classes like `RunCreationOptions` or other specific object used in the dotnet SDK. You can view the OpenAI doc [here](https://platform.openai.com/docs/api-reference/runs/createRun) or the AzureOpenAI doc [here](https://learn.microsoft.com/en-us/azure/ai-services/openai/assistants-reference-runs?tabs=python), as an example.

Reply:
Yes, this is an internal implementation detail of the .Net SDK (not visible to the caller).  The underlying service endpoint / network traffic should be identical and logically equivalent.